### ISSUE TITLE
pnpm cannot add dependency to workspace project on windows: Cannot destructure property 'manifest' of 'manifestsByPath[rootDir]' as it is undefined.

### ISSUE THREAD

**Issue by menasheh** [pnpm/pnpm#6524]
Trying to add a dependency while in a package's subdirectory of pnpm workspace:

### pnpm version:
8.4.0

### Code to reproduce the issue:
1. cd to subproject of a pnpm repo
2. `pnpm i vite-plugin-static-copy` (any package)

### Expected behavior:
pnpm should install the package in the current workspace project

### Actual behavior:
$ pnpm i vite-plugin-static-copy
 ERROR  Cannot destructure property 'manifest' of 'manifestsByPath[rootDir]' as it is undefined.

pnpm: Cannot destructure property 'manifest' of 'manifestsByPath[rootDir]' as it is undefined.
    at C:\NVM\nvm\versions\node\v18.12.1\bin\node_modules\pnpm\dist\pnpm.cjs:189340:19
    at async Promise.all (index 0)
    at async recursive (C:\NVM\nvm\versions\node\v18.12.1\bin\node_modules\pnpm\dist\pnpm.cjs:189337:9)
    at async installDeps (C:\NVM\nvm\versions\node\v18.12.1\bin\node_modules\pnpm\dist\pnpm.cjs:189727:11)
    at async C:\NVM\nvm\versions\node\v18.12.1\bin\node_modules\pnpm\dist\pnpm.cjs:216935:21
    at async main (C:\NVM\nvm\versions\node\v18.12.1\bin\node_modules\pnpm\dist\pnpm.cjs:216902:34)
    at async runPnpm (C:\NVM\nvm\versions\node\v18.12.1\bin\node_modules\pnpm\dist\pnpm.cjs:217157:5)
    at async C:\NVM\nvm\versions\node\v18.12.1\bin\node_modules\pnpm\dist\pnpm.cjs:217149:7

### Additional information:
 - `node -v` prints: v18.12.1
 - Windows, macOS, or Linux?: Windows, git bash and cmd same result.
 - may or may not be related to #4904
 - cd to workspace root and using --filter works

**Comment by menasheh** [comment]
@zkochan still not working after 8.5.1 release

**Comment by ksandin** [comment]
Happens to me as well using pnpm 8.7.1

**Comment by grimly** [comment]
The issue is still present on version `8.14.0`.

I managed to identify the issue.

I'll submit a PR soon enough.

**Comment by BearToCode** [comment]
I got this error after cancelling an installation, which probably broke something. I just had to restart my PC for it to start working again. Just an heads up to save someone a bit of time

**Comment by suingues** [comment]
Same issue with me on 9.1.4

**Comment by itsNMD404** [comment]
Have this issue on 9.1.3 no clue why

**Comment by Arcktosh** [comment]
Just started getting this error again on `9.4.0`

**Comment by JackatDJL** [comment]
Also on 9.4.0

**Comment by EthanQuarry** [comment]
Also on 9.6.0

**Comment by chrhi** [comment]
i got the same error  on 9.9.0

**Comment by JoseLeMalin** [comment]
Hello,
I have been trying to build a monorepo with pnpm version: 9.10.0, and the following repo structure:
![image](https://github.com/user-attachments/assets/3538b5c4-72f1-415f-a1bd-cdd78f373e7d)

But whenever I try to install a new package using the "pnpm i packagexyz" in one of the front or backend folder, I get the error message "Cannot destructure property 'manifest' of 'manifestsByPath[rootDir]' as it is undefined."

**Comment by misium** [comment]
pnpm v9.10.0, same issue.

[This issue](https://github.com/pnpm/pnpm/issues/4904) suggests it's a Windows issue (I'm Win 11 Home)

thanks to @BearToCode, a reboot just works:

> I just had to restart my PC for it to start working again. Just an heads up to save someone a bit of time

**Comment by grimly** [comment]
This issue is closed. Maybe you find the same error but that would be for a different cause.

I suggest you provide a constant minimal scenario causing your issue like I did with my PR.
You do not need to provide a fix, just provide the scenario and the error and an issue

**Comment by danmayor** [comment]
What did I miss?  I see many people reporting this problem for about 3 years now, across many versions.  And the solution is just (Issue is closed, try producing a minimal scenario)?

**Comment by usmangurowa** [comment]
Incase you are still having this issue:

Uninstall and install pnpm again

**Comment by catnoteafish** [comment]
the same issue as `10.24.0` version of pnpm

**Comment by AdamR33s** [comment]
Hello everyone, I seem to have found a workaround here by accident.

I hit the same error as shown above. I used pnpm to uninstall Prisma v6.19.0, then when trying to run `pnpm i prisma`, the above error was shown. To get around this I manually added the dependencies to the Package.json file and then just ran `pn i`. The error didn't happen and the prisma package installed correctly.

Hoping this might help!

**Comment by catnoteafish** [comment]
> Hello everyone, I seem to have found a workaround here by accident.
> 
> I hit the same error as shown above. I used pnpm to uninstall Prisma v6.19.0, then when trying to run `pnpm i prisma`, the above error was shown. To get around this I manually added the dependencies to the Package.json file and then just ran `pn i`. The error didn't happen and the prisma package installed correctly.
> 
> Hoping this might help!

Yep, it's work but not a good idea to solve it. With breakpoint debugging tools, I found that the path in user's profile directory will get different result. For example, if you create a new project in your desktop directory, it will return `$USERPROFILE/desktop/path/to/project` and `$USERPROFILE/Desktop/path/to/project` If you try to install a package, it will use the first one, that cause no directory matches candidate option and throw a accessing undefined error. However, I discovered it three months ago, and pnpm is so complicated that pushed me to stop to research, that I can't make sure there is something changing or the bug still happens, For me, just moving your project to any directory wihtout your **USERPROFILE DIRECTORY** is helpful.
